edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -86,9 +103,39 @@ internal static CreateContainerParameters GetCreateOptions(string createOptions)
         }
 
         internal static string ValidateAndGetImage(string image)
+        {
+            return ValidateAndGetImage(image, new EnvironmentWrapper());
+        }
+
+        internal static string ValidateAndGetImage(string image, IEnvironmentWrapper env)
         {
             image = Preconditions.CheckNonWhiteSpace(image, nameof(image)).Trim();
+
+            if (image.Substring(0, 1) == "$")
```

Comment:
```suggestion
            if (image[0] == '$')
```

Reply:
check image is not null or empty.

Reply:
As agreed during discussion, we are going to support $upstream only.  You should check if image starts with $upstream and replace it.

Reply:
1. Fixed
2. That is what I did. I separate, this way I can catch when an image is not correctly formated, Iike typo.